### PR TITLE
feat: improve in order linker wiring

### LINKED ISSUE #12660: Convert in order linker into a modernized component
The in order linker is currently a component in the wiring framework, but it is using old clunky APIs. Clean up use of this component so that it uses the proper non-deprecated APIs.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #12660

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/AbstractInOrderLinker.java [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+/**
+ * Links events to their parents. Expects events to be provided in topological order.
+ * <p>
+ * Will not link events to parents in the following cases:
+ * <ul>
+ *     <li>The parent is ancient</li>
+ *     <li>The parent's generation does not match the generation claimed by the child event</li>
+ *     <li>The parent's time created is greater than or equal to the child's time created</li>
+ * </ul>
+ * Note: This class doesn't have a direct dependency on the {@link Shadowgraph ShadowGraph},
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/AbstractInOrderLinker.java:57]: This comment is no longer strictly true for this base class. Perhaps it should be moved down into the `GossipLinker`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java [hunk 2]
```diff
@@ -128,212 +46,21 @@ public InOrderLinker(@NonNull final PlatformContext platformContext) {
...
-                && parentTimeCreated.compareTo(childTimeCreated) >= 0) {
-
-            childTimeIsNotAfterSelfParentTime(child, candidateParent, parentTimeCreated, childTimeCreated);
-
-            return null;
-        }
-
-        return candidateParent;
-    }
+    void clear(@NonNull final NoInput ignored);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java:65]: optional: add wire annotation

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/AbstractInOrderLinker.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/AbstractInOrderLinker.java:57]: moved

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/linking/InOrderLinker.java:65]: The default label for an input wire is the name of the method. In this case, the name of the method is the name I want to have for this wire.

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/generator/StandardGraphGenerator.java [hunk 3]
```diff
@@ -201,11 +196,12 @@ private StandardGraphGenerator(final StandardGraphGenerator that, final long see
 
     private void initializeInternalConsensus() {
         consensus = new ConsensusImpl(platformContext, new NoOpConsensusMetrics(), addressBook);
-        inOrderLinker = new InOrderLinker(platformContext);
+        inOrderLinker = new ConsensusLinker(platformContext, new NodeId(0));
```
Comment (kfa-aguda) [inline platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/generator/StandardGraphGenerator.java:199]: probably obvious but why is this node(0), and why is it hardcoded?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java [hunk 4]
```diff
@@ -464,4 +467,36 @@ public EventCreationManager buildEventCreationManager() {
...
     }
+
+    /**
+     * Provide an in-order linker in place of the platform's default in-order linker.
+     *
+     * @param inOrderLinker the in-order linker to use
+     * @return this builder
+     */
+    @NonNull
+    public PlatformComponentBuilder withInOrderLinker(@NonNull final InOrderLinker inOrderLinker) {
```
Comment (kfa-aguda) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:478]: unused method.

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java (hunk 4 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformComponentBuilder.java:478]: This is used. See `PlatformWiringTests.java`.

File: platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/generator/StandardGraphGenerator.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/event/generator/StandardGraphGenerator.java:199]: This field is only used for metrics, which are disabled when running with this framework. The value we pass into the constructor does not matter.

**Pull request merged**